hawc/apps/lit/managers.py

```diff
@@ -295,6 +295,41 @@ def with_tag(self, tag, descendants: bool = False):
             tag_ids.extend(list(tag.get_descendants().values_list("pk", flat=True)))
         return self.filter(tags__in=tag_ids).distinct("pk")
 
+    def require_tags(self, tags):
+        """
+        Filter references by any of the given tags.
+
+        Mirrors the filtering done in tagtree visuals.
+
+        Args:
+            tags: Required tags.
+        """
+        query = Q()
+        for tag in tags:
+            query |= Q(tags__in=[tag, *tag.get_descendants()])
```

Comment:
haven't completely gotten through this PR yet, but I did find a case where if a user filters by two items in two different trees, it shows both. I think it should be some sort of `&=` instead of `|=`, but it's more complicated than a quick change.

Reply:
I tried this, and it would work, but if you're also excluding then you get an error:

```python
matches = []
for tag in tags:
    matches.append(self.filter(tags__in=[tag, *tag.get_descendants()]))
return self.intersection(*matches)
```

Reply:
got a little further, but now some other numbers aren't adding up

```python
def require_tags(self, tags, lazy: bool = True):
        """
        Filter references by any of the given tags.

        Mirrors the filtering done in tagtree visuals.

        Args:
            tags: Required tags.
            lazy: Do not evaluate the queryset (intersection can cause issues downstream)
        """

        matches = []
        for tag in tags:
            matches.append(self.filter(tags__in=[tag, *tag.get_descendants()]))
        qs = self.intersection(*matches)
        if lazy:
            return qs
        ids = list(qs.values_list("id", flat=True))
        return self.filter(id__in=ids)
```

Reply:
Per our discussion, this approach would work for the case I described, but that is not consistent with our tag tree filtering; it currently DOES do an or by all tags, but does not include descendants of the tags, which is likely why we were getting incorrect behavior. 

So, the change, as suggested by @rabstejnek, is likely to remove the descendant checking from the original QuerySet operation

Reply:
I parameterized `descendants` in these manager methods, and it looks like `descendants=False` for the required tags does fix the problems we were seeing.